packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -242,13 +244,25 @@ class ${class_name}:
     def _deserialize_response(
         self, response: v2_measurement_service_pb2.MeasureResponse
     ) -> Outputs:
+        self._validate_response(response)
         return Outputs._make(
             deserialize_parameters(
                 self._output_metadata,
                 response.outputs.value,
                 f"{self._service_class}.Outputs",
             )
         )
+
+    def _validate_response(
+        self, response: v2_measurement_service_pb2.MeasureResponse
+    ) -> None:
+        expected_type = f"type.googleapis.com/{self._service_class}.Outputs"
```

Comment:
This should expect the advertised message type, not the message type we made up for the descriptor pool. For Python services, these are now the same thing, but for LV and C# services they aren't. Please generate a client for a C# service and manually test it. (You can load the client in the REPL and call update_pin_map() and measure().)